server/src/main/java/org/opensearch/search/profile/query/ConcurrentQueryProfileBreakdown.java

```diff
@@ -235,13 +251,28 @@ Map<Collector, Map<String, Long>> buildSliceLevelBreakdown() {
                 sliceMinStartTime = Math.min(sliceMinStartTime, currentSliceStartTime);
                 // compute total time for each timing type at slice level using sliceEndTime and sliceStartTime
                 currentSliceBreakdown.put(
-                    timingType.toString(),
+                    timingTypeTimeKey,
                     currentSliceBreakdown.getOrDefault(timingTypeSliceEndTimeKey, 0L) - currentSliceBreakdown.getOrDefault(
                         timingTypeSliceStartTimeKey,
                         0L
                     )
                 );
             }
+
+            for (String metric : nonTimingMetrics) {
+                for (LeafReaderContext sliceLeaf : slice.getValue()) {
```

Comment:
I think this is probably fine, but we have to iterate over slice leafs two times: once for timing metrics, and once for non-timing ones. This is probably fine, profiling is always opt-in.